src/Microsoft.ML.Featurizers/DateTimeTransformer.cs

```diff
@@ -0,0 +1,782 @@
+
+            [Argument(ArgumentType.MultipleUnique, HelpText = "Columns to drop after the DateTime Expansion", Name = "ColumnsToDrop", ShortName = "drop", SortOrder = 3)]
+            public ColumnsProduced[] ColumnsToDrop;
+
+            [Argument(ArgumentType.AtMostOnce, HelpText = "Country to get holidays for. Defaults to none if not passed", Name = "Country", ShortName = "ctry", SortOrder = 4)]
+            public Countries Country = Countries.None;
+        }
+
+        #endregion
+
+        public DateTimeTransformerEstimator(IHostEnvironment env, string inputColumnName, string columnPrefix, ColumnsProduced[] columnsToDrop, Countries country = Countries.None)
+        {
+
+            Contracts.CheckValue(env, nameof(env));
+            _host = Contracts.CheckRef(env, nameof(env)).Register("DateTimeTransformerEstimator");
+            _host.CheckValue(inputColumnName, nameof(inputColumnName), "Input column should not be null.");
+
+            _options = new Options
+            {
+                Source = inputColumnName,
+                Prefix = columnPrefix,
+                ColumnsToDrop = columnsToDrop == null ? Array.Empty<ColumnsProduced>() : columnsToDrop,
+                Country = country
+            };
+        }
+
+        internal DateTimeTransformerEstimator(IHostEnvironment env, Options options)
+        {
+
+            Contracts.CheckValue(env, nameof(env));
+            _host = Contracts.CheckRef(env, nameof(env)).Register("DateTimeTransformerEstimator");
+
+            _options = options;
+            _options.ColumnsToDrop = _options.ColumnsToDrop == null ? Array.Empty<ColumnsProduced>() : _options.ColumnsToDrop;
+        }
+
+        public DateTimeTransformer Fit(IDataView input)
+        {
+            return new DateTimeTransformer(_host, _options.Source, _options.Prefix, _options.ColumnsToDrop, _options.Country);
+        }
+
+        public SchemaShape GetOutputSchema(SchemaShape inputSchema)
+        {
+            var columns = inputSchema.ToDictionary(x => x.Name);
+
+            foreach (ColumnsProduced column in Enum.GetValues(typeof(ColumnsProduced)))
+                if (_options.ColumnsToDrop == null || !_options.ColumnsToDrop.Contains(column))
+                {
+                    columns[_options.Prefix + column.ToString()] = new SchemaShape.Column(_options.Prefix + column.ToString(), SchemaShape.Column.VectorKind.Scalar,
+                    ColumnTypeExtensions.PrimitiveTypeFromType(column.GetRawColumnType()), false, null);
+                }
+
+            return new SchemaShape(columns.Values);
+        }
+
+        #region Enums
+        public enum ColumnsProduced : byte
+        {
+            Year = 1, Month, Day, Hour, Minute, Second, AmPm, Hour12, DayOfWeek, DayOfQuarter, DayOfYear,
+            WeekOfMonth, QuarterOfYear, HalfOfYear, WeekIso, YearIso, MonthLabel, AmPmLabel, DayOfWeekLabel,
+            HolidayName, IsPaidTimeOff
+        };
+
+        public enum Countries : byte
```

Comment:
Is using an `enum` for countries the best route here? And if so, do you want this to be a `[Flags]` enum so you can request multiple countries?

Reply:
So currently you can only request the holidays for one country at a time, since we can only return one holiday name and one paid time off value, so I think that an enum is fine for that.

---
In reply to: [353907829](https://github.com/dotnet/machinelearning/pull/4521#discussion_r353907829) [](ancestors = 353907829)

Reply:
What is the purpose of returning a holiday name (string)? Shouldn't we return a boolean of `IsHoliday`? Or perhaps the boolean and a one-hot encoded version of which holiday? The slot names of the one-hot encoded field can be the exact holiday name.

In the end, all of the created features need to be turned into a float32, as the model takes in a float32 array as a feature vector.

You may also want to include `TimeUntilNextHoliday` and `TimeLastLastHoliday`. @EricWrightAtWork, which DateTime features are needed for AutoML forecasting?